tools/CheckOutputFiles.py

```diff
@@ -0,0 +1,230 @@
+# Distributed under the MIT License.
```

Comment:
To get this script running with current develop (and address your very reasonable aversion to merging code that doesn't run on CI) I suggest you add an `execute_check_h5` to an existing input file. I'm sure one of the evolution input files would work fine, but here's a suggested diff to get this working with one of the elliptic input files (since I'm more familiar with those):

```diff
diff --git a/tests/InputFiles/Poisson/ProductOfSinusoids1D.yaml b/tests/InputFiles/Poisson/ProductOfSinusoids1D.yaml
index 6725ea8cf..93cc0e271 100644
--- a/tests/InputFiles/Poisson/ProductOfSinusoids1D.yaml
+++ b/tests/InputFiles/Poisson/ProductOfSinusoids1D.yaml
@@ -2,10 +2,15 @@
 # See LICENSE.txt for details.
 
 # Executable: SolvePoissonProductOfSinusoids1D
-# Check: parse;execute
+# Check: parse;execute;execute_check_h5
 # ExpectedOutput:
 #   PoissonProductOfSinusoids1DReductions.h5
 #   PoissonProductOfSinusoids1DVolume0.h5
+# OutputFileChecks:
+#   - Label: Discretization error
+#     Subfile: /Errors.dat
+#     FileGlob: PoissonProductOfSinusoids1DReductions.h5
+#     AbsoluteTolerance: 0.02
 
 AnalyticSolution:
   ProductOfSinusoids:
@@ -30,15 +35,15 @@ Observers:
 
 LinearSolver:
   ConvergenceCriteria:
-    MaxIterations: 1
-    AbsoluteResidual: 0
-    RelativeResidual: 0
+    MaxIterations: 7
+    AbsoluteResidual: 1e-10
+    RelativeResidual: 1e-10
   Verbosity: Verbose
 
 EventsAndTriggers:
   ? EveryNIterations:
       N: 1
-      Offset: 0
+      Offset: 6
   : - ObserveErrorNorms:
         SubfileName: Errors
     - ObserveFields:
```